src/webfaf/blueprints/celery_tasks.py

```diff
@@ -286,8 +403,11 @@ def action_run(action_name):
     pt_form = PeriodicTaskForm(request.form)
     if request.method == "POST":
         if request.values.get("run") and action_form.validate():
-            run_action.delay(action_name, action_form.to_cmdline_dict())
-            flash("Action {0} started successfully.".format(action_name), "success")
+            output = run_action(action_name, action_form.to_cmdline_dict())
```

Comment:
What's the reason for running the task in-process instead of having a worker do it? This will likely lead to timeouts for long-running tasks.

Reply:
Honestly, I have no idea what to do with this. The only change I did here was to remove the `delay` method(?), which magically made the actions work. (Before that, clicking the 'Run once' button in the web UI didn't actually do anything.) I can't find any similar uses of `delay` in the code or any pointers as to what it does. @ernestask , @mkutlak, any ideas?

Reply:
https://docs.celeryproject.org/en/latest/reference/celery.app.task.html#celery.app.task.Task.delay

Seems that it runs the task asynchronously, IIUC.

Reply:
Solved. The original code was actually correct, only it was relying on the faf-celery-worker service in the host, which wasn't running.